float32  performanceCost
float32  resourceCost
